api/server/router/image/image_routes.go

```diff
@@ -215,15 +215,15 @@ func (ir *imageRouter) postImagesPush(ctx context.Context, w http.ResponseWriter
 		if formPlatform := r.Form.Get("platform"); formPlatform != "" {
 			p, err := httputils.DecodePlatform(formPlatform)
 			if err != nil {
-				return err
+				return errdefs.InvalidParameter(err)
 			}
 			platform = p
 		}
 	}
 
 	if err := ir.backend.PushImage(ctx, ref, platform, metaHeaders, authConfig, output); err != nil {
 		if !output.Flushed() {
-			return err
+			return errdefs.InvalidParameter(err)
```

Comment:
it looks like the backend PushImage already returns a typed error, and we can't assume it's always an invalid parameter here, so this should probably not be changed https://github.com/moby/moby/blob/980d8e5d54d8df31ba0e75e1d9dd474d9d91308a/daemon/images/image_push.go#L21